feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch

```diff
@@ -0,0 +1,1489 @@
++            break;
++
+         default:
+             LOG(ERR, "Failed to add %d to stats report", s->type);
+             /* do nothing       */
+@@ -2106,6 +2334,14 @@ bool dpp_put_network_probe(dpp_network_p
+ }
+ 
+ /*
++ * Put event report to internal queue
++ */
++bool dpp_put_events(dpp_event_report_data_t *rpt)
++{
++    return dppline_put(DPP_T_EVENTS, rpt);
++}
++
++/*
+  * Create the protobuf buff and copy it to given buffer
+  */
+ #ifndef DPP_FAST_PACK
+Index: opensync-2.0.5.0/platform/openwrt/src/lib/target/inc/target_OPENWRT.h
+===================================================================
+--- opensync-2.0.5.0.orig/platform/openwrt/src/lib/target/inc/target_OPENWRT.h
++++ opensync-2.0.5.0/platform/openwrt/src/lib/target/inc/target_OPENWRT.h
+@@ -25,10 +25,11 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
+ */
+ 
+ #ifndef TARGET_OPENWRT_H_INCLUDED
+-#define TARGET_OPEMWRT_H_INCLUDED
++#define TARGET_OPENWRT_H_INCLUDED
+ 
+ #include "dpp_client.h"
+ #include "dpp_survey.h"
++#include "dpp_events.h"
+ 
+ #define TARGET_CERT_PATH            "/var/run/openvswitch/certs"
+ #define TARGET_MANAGERS_PID_PATH    "/tmp/dmpid"
+@@ -54,6 +55,12 @@ typedef struct
+ 	uint32_t duration_ms;
+ } target_survey_record_t;
+ 
++typedef struct
++{
++        dpp_event_record_t record;
++
++} target_event_record_t;
++
+ typedef void target_capacity_data_t;
+ 
+ /******************************************************************************
+Index: opensync-2.0.5.0/platform/openwrt/src/lib/target/src/stats.c
+===================================================================
+--- opensync-2.0.5.0.orig/platform/openwrt/src/lib/target/src/stats.c
++++ opensync-2.0.5.0/platform/openwrt/src/lib/target/src/stats.c
+@@ -66,6 +66,43 @@ bool target_radio_fast_scan_enable(radio
+ 	return true;
+ }
+ 
++/******************************************************************************
++ *  EVENT definitions
++ *****************************************************************************/
++target_event_record_t* target_event_record_alloc()
++{
++	target_event_record_t *record = NULL;
++
++	record = malloc(sizeof(target_event_record_t));
++	if (record == NULL)
++		return NULL;
++
++	memset(record, 0, sizeof(target_event_record_t));
++
++	return record;
++}
++
++void target_event_record_free(target_event_record_t *record)
++{
++	if (record != NULL)
++		free(record);
++}
++
++bool target_stats_events_get(radio_entry_t *radio_cfg,
++                             ds_dlist_t *record_list)
++{
++	struct nl_call_param nl_call_param = {
++		.ifname = radio_cfg->if_name,
++		.type = radio_cfg->type,
++                .list = record_list
++	};
++	bool ret = true;
++	if (nl80211_rt_events_handler(&nl_call_param) < 0)
++		ret = false;
++
++        return ret;
++}
++
+ 
+ /******************************************************************************
+  *  CLIENT definitions
+Index: opensync-2.0.5.0/platform/openwrt/src/lib/target/src/stats_nl80211.c
```

Comment:
dont patches files in the src/ folder but apply the patch to the actual file please